xarray/core/netcdftimeindex.py

```diff
@@ -0,0 +1,180 @@
+import re
+from datetime import timedelta
+
+import numpy as np
+import pandas as pd
+
+from pandas.lib import isscalar
+
+
+def named(name, pattern):
+    return '(?P<' + name + '>' + pattern + ')'
```

Comment:
I think `.format` is faster (as well as idiomatic) because this way will build n strings